hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/SnapshotInfo.java

```diff
@@ -71,8 +89,14 @@ public static SnapshotStatus valueOf(SnapshotStatusProto status) {
             "BUG: missing valid SnapshotStatus, found status=" + status);
       }
     }
-  };
+  }
 
+  public static final String SNAPSHOT_FLAG = "snapshot";
+  private static final String SEPARATOR = "-";
+  private static final long UNDELETED_TIME = -1;
+  private static final String INITIAL_SNAPSHOT_ID =
+      UUID.randomUUID().toString();
```

Comment:
instead of random UUID, can we have the timestamp based name ? we can list the snapshots in the order they were created.

Reply:
The snapshot name is a part of the key.  If the users chooses not to give a name, then the timestamp based name is generated.

But the only thing that keeps that name unique is that it is checked as a part of the key during the create: https://github.com/GeorgeJahad/ozone/blob/HDDS-6853-rebased2/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotCreateRequest.java#L128-L132

So if the user does select a non-timestamped based name, the timestamp one isn't used and thus isn't checked for uniqueness.

We could add a check for it, but that would require reading the entire snapshot info table.  Neil's PR with the snapshot chain would help with that.  Should we leave this for after that get merged in?